lib/cache-driver.js

```diff
@@ -243,6 +243,11 @@ CacheDriver.prototype._loadSingleEntryFromCache = function(query, callback) {
 CacheDriver.prototype._createEntryFromRefresh = function(entry, refreshResponse) {
   var newEntry = _.clone(entry);
   newEntry = _.extend(newEntry, refreshResponse);
+
+  if (entry.isMRRT && this._authority !== entry[METADATA_AUTHORITY]) {
+    newEntry[METADATA_AUTHORITY] = this._authority;
```

Comment:
Not very familiar with the code, but should we also update the "tenantId" field? Hope this help.